### PR TITLE
Make e2e -up recreate running clusters; remove kube-up retry

### PR DESCRIPTION
**Pull request by pmorie**

### CODE REVIEW

**Comment by pmorie** [comment]
Caveat; I coded this but haven't been able to test it yet personally due to equipment failure.

**Comment by pmorie** [comment]
@derekwaynecarr @deads2k @lavalamp PTAL

**Comment by zmerlynn** [comment]
Yeah, I was been meaning to get around to this change after adding `IsUp`. There's a slight semantic shift, though, which is why I was resisting it: Right now `IsUp` can easily be false, and yet k8s might have random resources left strewn around GCE that need to be cleaned up by teardown.

**Comment by zmerlynn** [comment]
I think the way to implement this change correctly for "fast" detection is to leave the original logic in, but also add a check at the top that does what you're doing. i.e.: If up, we know we need to teardown. Then try the kube-up. If that fails, and we haven't already tried a teardown, we do a teardown.

**Comment by lavalamp** [comment]
Is there any benefit to trying twice? It should always work, and if it doesn't, that's a problem...

**Comment by pmorie** [comment]
@zmerlynn I'm okay with what you suggest for keeping the retry logic while making an explicit check to see if the cluster's running, but I want the ability to turn off the retry.  It's annoying if you're debugging why a provider doesn't come up.

@lavalamp If it should always work, retrying is the last thing you should do :)

My problem with the retry is that it destroys the state that didn't work.

**Comment by lavalamp** [comment]
Yeah, for those reasons I think we should remove the retry. It was only there in the first place because there was originally no way to tell if the cluster was up.

**Comment by zmerlynn** [comment]
That's fine. Jenkins actually calls `-down` explicitly in a single call prior to calling `-up`, so it's not going going to get stuck there either: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/hack/jenkins/e2e.sh

This actually means cluster creation build failures will be faster, since we won't retry. So :+1:

**Comment by pmorie** [comment]
@zmerlynn @lavalamp I will test this once my new hardware arrives.  Unless I'm misreading you both, it sounds like we need no further changes.

**Comment by zmerlynn** [comment]
This change is fine, pending testing (of a cloud provider ... there's some
irony there).

File: hack/e2e.go [hunk 1]
```diff
@@ -184,13 +184,14 @@ func TearDown() {
 	runBash("teardown", "test-teardown")
 }
 
+// Up brings an e2e cluster up, recreating it if one is already running.
 func Up() bool {
-	if !tryUp() {
-		log.Printf("kube-up failed; will tear down and retry. (Possibly your cluster was in some partially created state?)")
+	if isUp() {
+		log.Printf("e2e cluster already running; will teardown")
```
Comment (lavalamp) [inline hack/e2e.go]: Only one suggestion-- perhaps should fail if this TearDown fails, rather than calling up again...

Comment (pmorie) [inline hack/e2e.go]: I think it should, good point.

**Comment by zmerlynn** [comment]
@pmorie: Still pending testing and the change above, yes?

**Comment by pmorie** [comment]
@zmerlynn Yep, still pending.  Eta would be tomorrow afternoon.

**Comment by pmorie** [comment]
@zmerlynn @lavalamp I got a successful e2e run in master on gce this morning; have plans today but hopefully will test this tonight or tomorrow am.

**Comment by pmorie** [comment]
@zmerlynn @lavalamp PTAL; I've tested on GCE.

**Comment by zmerlynn** [comment]
LGTM.

**Pull request merged**